### PR TITLE
GRF-233 : Add value collection in query sql to save a base category

### PR DESCRIPTION
**Pull request by antoinetrouve**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by BryanS29-akeneo — APPROVED** [review]

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php [hunk 1]
```diff
@@ -45,12 +45,17 @@ private function insertCategory(Category $categoryModel): void
...
             INSERT INTO pim_catalog_category
-                (parent_id, code, created, root, lvl, lft, rgt)
+                (parent_id, code, created, root, lvl, lft, rgt, value_collection)
             VALUES
-                (:parent_id, :code, NOW(), :root, :lvl, :lft, :rgt)
+                (:parent_id, :code, NOW(), :root, :lvl, :lft, :rgt, :value_collection)
             ;
         SQL;
 
+        $values = $categoryModel->getAttributes();
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php:54]: Here you have to respect the format of the JSON format laid out in the [ADR](https://www.notion.so/akeneo/Enrich-Categories-flexibility-and-maintenance-of-the-data-23b99c6ac4af4997b8c307f4e94291ca#bf6ab26e7d9d4f109c1b6f55fdbb27f9). You need to have a table of composite keys at the start of the JSON before the list of attributes concatenated with their locale codes

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php [hunk 2]
```diff
@@ -104,20 +111,27 @@ private function updateCategory(Category $categoryModel): void
...
                     lvl = :lvl,
                     lft = :lft,
-                    rgt = :rgt
+                    rgt = :rgt,
+                    value_collection = :value_collection
                 WHERE code = :category_code
                 ;
             SQL;
 
+        $values = $categoryModel->getAttributes();
```
Comment (etiennekerbastard) [inline src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php:120]: same here

**Review by etiennekerbastard — CHANGES_REQUESTED** [review]

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php (hunk 1 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php:54]: Yes this is the role of the handler to manage data format. As per discussion i add some test to validate the data and their values saved in db.

File: src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php (hunk 2 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/UpsertCategoryBaseSql.php:120]: same here ;)

**Review by etiennekerbastard — APPROVED** [review]

**Pull request merged**